### PR TITLE
[feat]: [PL-28744]: updating default onDelegate value to true

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Describe your changes
Changing default value of OnDelegate to true. 
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38371)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38371)

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38371)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by shreyas-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38371)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/customsecretmanager/CustomSecretManagerConnector.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/38371#-NDaFaZbBd_GegO0bJiu:-NDaG_3nB5coRTFb6byU:b-goni1s) ([raw file](https://github.com/harness/harness-core/blob/8ecb618e4643064a02648a8b1f5bb7f7228f6e08/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/customsecretmanager/CustomSecretManagerConnector.java#L47)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

What will happen in case, where existing doc in the collection doesn't have this field. Will this be set as null? If that is the case then getter method should take care of returning the correct value.
</blockquote></details>

We are working on writing migrator to add it to all existing connectors.

**Comment by shreyas-harness** [comment]
trigger messagecheck

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38371)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java` line 93 at r2](https://reviewable.io/reviews/harness/harness-core/38371#-NDfO2ghEw68Sq7XVyEK:-NDfO2ghEw68Sq7XVyEL:bh49ajp) ([raw file](https://github.com/harness/harness-core/blob/7151d60be34d4241effb7cef6abf2faf02af4471/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java#L93)):*
> ```Java
>             errorMessageBuilder.toString(), ErrorCode.INVALID_REQUEST, WingsException.USER);
>       }
>     } else {
> ```

break into methods.
Also we should have UT covering validation

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 1]
```diff
@@ -63,4 +67,46 @@ public class CustomSecretManagerConnectorDTO extends ConnectorConfigDTO implemen
   public List<DecryptableEntity> getDecryptableEntities() {
     return Collections.singletonList(this);
   }
+
+  @Override
+  public void validate() {
+    if (Boolean.TRUE.equals(onDelegate)) {
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:73]: I don't think this validation is needed. Even if user passes extra info. We should drop it.. if onDelegate is true.
But if onDelegate is false, we need some info (host etc) to be not null. For this we can have validation. What do you think?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 2]
```diff
@@ -63,4 +67,46 @@ public class CustomSecretManagerConnectorDTO extends ConnectorConfigDTO implemen
...
+        throwException = true;
+        errorMessageBuilder.append(String.format("working directory = %s ", workingDirectory));
+      }
+      if (throwException) {
+        throw new InvalidRequestException(
+            errorMessageBuilder.toString(), ErrorCode.INVALID_REQUEST, WingsException.USER);
+      }
+    } else {
+      List<String> targetMachineNullFields = new LinkedList<>();
+      if (host == null) {
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:95]: Should we use `isEmpty` ?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 3]
```diff
@@ -63,4 +67,46 @@ public class CustomSecretManagerConnectorDTO extends ConnectorConfigDTO implemen
...
+        targetMachineNullFields.add("host");
+      }
+      if (workingDirectory == null) {
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:98]: same

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 4]
```diff
@@ -63,4 +67,46 @@ public class CustomSecretManagerConnectorDTO extends ConnectorConfigDTO implemen
...
+        targetMachineNullFields.add("working directory");
+      }
+      if (connectorRef == null) {
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:101]: same

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 5]
```diff
@@ -63,4 +67,46 @@ public class CustomSecretManagerConnectorDTO extends ConnectorConfigDTO implemen
...
+        targetMachineNullFields.add("connector ref");
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:102]: Should be `connectorRef` ?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Pull request closed without merging**